pkg/email/patch.go

```diff
@@ -6,33 +6,36 @@ package email
 import (
 	"bufio"
 	"fmt"
+	"regexp"
 	"strings"
 )
 
 func ParsePatch(text string) (title string, diff string, err error) {
 	s := bufio.NewScanner(strings.NewReader(text))
-	parsingDiff := false
-	diffStarted := false
 	lastLine := ""
+	diffStarted := false
 	for s.Scan() {
 		ln := s.Text()
-		if strings.HasPrefix(ln, "--- a/") || strings.HasPrefix(ln, "--- /dev/null") {
-			parsingDiff = true
+		if lineMatchesDiffStart(ln) {
+			diffStarted = true
+			diff += ln + "\n"
 			if title == "" {
 				title = lastLine
 			}
+			continue
 		}
-		if parsingDiff {
-			if ln == "" || ln == "--" || ln == "-- " || ln[0] == '>' ||
-				ln[0] >= 'A' && ln[0] <= 'Z' {
-				break
+		if diffStarted {
+			if ln == "" || ln == "--" || ln == "-- " || ln[0] == '>' {
+				diffStarted = false
+				continue
 			}
-			diff += ln + "\n"
-			continue
 		}
-		if strings.HasPrefix(ln, "diff --git") {
-			diffStarted = true
-			continue
+		if diffStarted {
```

Comment:
Please merge this with this the previous if, it checks the same condition.

Reply:
Done.